File: FrontEnd/src/app/models/products/product.ts
import { Colour } from './colour';
import { ProductType } from './ProductType';
import { Size } from './size';

export interface Product {
  productId: number;
  productName: string;
  productDescription: string;
  productBarcode: string;
  quantityAvailable: number;
  productPicture: string;
  productPrice: number;
  productType: ProductType;
  sizes: Size[];
  colours: Colour[];
}

File: SheCollectionBE/SheCollectionBE/DB Models/Product.cs
using System;
using System.Collections.Generic;

namespace SheCollectionBE.Models
{
    public partial class Product
    {
        public int ProductId { get; set; }
        public int ProductTypeId { get; set; }
        public string ProductName { get; set; }
        public string ProductDescription { get; set; }
        public string ProductBarcode { get; set; }
        public int QuantityAvailable { get; set; }
        public string ProductPicture { get; set; }
        public decimal ProductPrice { get; set; }

        public virtual ProductType ProductType { get; set; }
    }
}
